plugins/ingest-attachment/src/main/java/org/elasticsearch/ingest/attachment/AttachmentProcessor.java

```diff
@@ -119,7 +119,12 @@ public void execute(IngestDocument ingestDocument) {
 
             if (properties.contains(Property.CONTENT_LENGTH)) {
                 String contentLength = metadata.get(Metadata.CONTENT_LENGTH);
-                String length = Strings.hasLength(contentLength) ? contentLength : String.valueOf(parsedContent.length());
+                long length;
+                if (Strings.hasLength(contentLength)) {
+                    length = Long.parseLong(contentLength);
```

Comment:
> And there's the ability to attach an on error processor?

That'd mean throwing out all the data we extract because of one bad header. Maybe that is ok?

I get the sense that we should be lenient with documents and metadata fields because I don't think they really have a specification. Or maybe not a common one. But I'm shooting from the hip here, just going on instinct based on working with POI years and years ago.

Another option is to simply not add a the field at all if it isn't a number. But that feels weird because we already add the field if it isn't present.